### PR TITLE
Fixed typos in test doubles

### PR DESCRIPTION
**Pull request by abxli**
As the title says, fixed a few typos in test doubles.

### CODE REVIEW

File: chapters/pragmatic-testing/test-doubles.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ We'll be showing some examples of how this works later in the chapter.
...
+* Firstly, we have **more control**. We can easily tell these objects what to do, without the need for complicated setups. If we want a method to throw an exception, we just tell the mock method to throw it; there is no need for complicated setups to "force" the dependency to throw the exception. Similarly, if we want a method that returns a Calendar object representing a date in 2009, we just tell the mock method to do it; there is no need to complicate your OS in order to make it go back in time. Note how hard it usually is to "force" a class to throw an exception, or to "force" a class to return a fake date. This effort is close to zero when we simulate the dependencies with mock objects.
```
Comment (abxli) [inline chapters/pragmatic-testing/test-doubles.md:18]: I added a comma after firstly ( don't know why it doesn't highlight the change)

File: chapters/pragmatic-testing/test-doubles.md [hunk 2]
```diff
@@ -465,19 +465,19 @@ Hence, once you get used to this way of developing, you will notice how your cod
...
+In their seminal work _Mock Roles, Not Objects_, Freeman et al. show how Mock Objects can be used for thinking about _design_ as much as for testing. By treating an object (or a cluster of objects) as a closed box, the programmer can focus on how it interacts with its environment: if the object receives a triggering event (one or more method calls), what does it need to find out (stubs) and how does it change the world around it (mocks)? This, in turn, means that the programmer needs to make explicit what else the target object depends on. Because these tests focus on how an object interacts with its environment, this style of testing is sometimes called "Interaction Testing."
```
Comment (abxli) [inline chapters/pragmatic-testing/test-doubles.md:468]: I changed "calls" to "called", again I don't know why it doesn't highlight the change.

File: chapters/pragmatic-testing/test-doubles.md [hunk 3]
```diff
@@ -465,19 +465,19 @@ Hence, once you get used to this way of developing, you will notice how your cod
...
+Interaction testing is most effective when combined with TDD. While writing a test, the programmer has to define what the object needs from its environment. For a new feature, this might require a new collaborating object, which introduces a new dependency. The type of this new dependency is defined in terms of what the target object needs - its caller, not its implementation; in Java, this is usually an interface. Following this process, with regular refactoring, a programmer can grow a codebase written in the terminology of the domain. The result is a set of pluggable objects, with clear dependencies, that are combined to build a system.
```
Comment (abxli) [inline chapters/pragmatic-testing/test-doubles.md:472]: I changed "refactorings" to "refactoring".

File: chapters/pragmatic-testing/test-doubles.md [hunk 4]
```diff
@@ -465,19 +465,19 @@ Hence, once you get used to this way of developing, you will notice how your cod
...
 
 The authors summarized their best practices for _interaction testing_ as:
 
 * *Only mock types you own.* Developers should not mock objects that they do not "own", e.g., a class from an external library. Rather, they should build abstractions on top of those.
 
 * *Be explicit about things that should not happen*. Writing tests for interactions that should not happen makes intentions clearer.
 
-* *Specify as little as possible in the test*. Overspecified tests tend to be more brittle. Focus on the interactions that really matter.
+* *Specify as little as possible in the test*. Over specified tests tend to be more brittle. Focus on the interactions that really matter.
```
Comment (mauricioaniche) [inline chapters/pragmatic-testing/test-doubles.md:480]: I think this should be a single word, right ?

File: chapters/pragmatic-testing/test-doubles.md [hunk 5]
```diff
@@ -510,7 +510,7 @@ The "Software Engineering at Google" book has an entire chapter dedicated to tes
 * Excessive mocking can be dangerous, as tests become unclear (i.e., hard to comprehend), brittle (i.e., might break too often), and less effective (i.e., reduced fault capability detection).
 * When mocking, prefer _state testing_ rather than _interaction testing_. In other words, make sure you are asserting a change of state and/or the consequence of the action under test, rather than the precise interaction that the action has with the mocked object. After all, interaction testing tends to be too coupled with the implementation of the system under test.
 * Use _interaction testing_ when state testing is not possible, or when a bad interaction might have an impact in the system (e.g., calling the same method twice would make the system twice as slow).
-* Avoid overspecified interaction tests. Focus on the relevant arguments and functions.
+* Avoid over specified interaction tests. Focus on the relevant arguments and functions.
```
Comment (mauricioaniche) [inline chapters/pragmatic-testing/test-doubles.md:513]: Same here. Are you sure we have to split it ?

File: .idea/workspace.xml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mauricioaniche) [inline .idea/workspace.xml:1]: Maybe you added this file as a mistake?

Comment (abxli) [reply .idea/workspace.xml:1]: Yes, sorry!

File: chapters/pragmatic-testing/test-doubles.md (hunk 5 above)
Comment (abxli) [reply chapters/pragmatic-testing/test-doubles.md:513]: Oh my bad, you can ignore this :)

File: chapters/pragmatic-testing/test-doubles.md (hunk 4 above)
Comment (abxli) [reply chapters/pragmatic-testing/test-doubles.md:480]: Yes, you're right. My bad.

**Pull request merged**